src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -223,15 +224,37 @@ public static ReadOnlySpan<char> GetFileNameWithoutExtension(ReadOnlySpan<char>
         /// Returns a cryptographically strong random 8.3 string that can be
         /// used as either a folder name or a file name.
         /// </summary>
-        public static unsafe string GetRandomFileName()
+        public static string GetRandomFileName()
         {
-            byte* pKey = stackalloc byte[KeyLength];
-            Interop.GetRandomBytes(pKey, KeyLength);
+            Span<char> buffer = stackalloc char[RandomFileNameLength];
+            bool result = TryGetRandomFileName(buffer, out int charsWritten);
 
-            const int RandomFileNameLength = 12;
-            char* pRandomFileName = stackalloc char[RandomFileNameLength];
-            Populate83FileNameFromRandomBytes(pKey, KeyLength, pRandomFileName, RandomFileNameLength);
-            return new string(pRandomFileName, 0, RandomFileNameLength);
+            Debug.Assert(charsWritten == RandomFileNameLength);
+            Debug.Assert(result);
+
+            return new string(buffer);
+        }
+
+        /// <summary>
+        /// Returns a cryptographically strong random 8.3 string that can be
+        /// used as either a folder name or a file name.
+        /// </summary>
+        public static unsafe bool TryGetRandomFileName(Span<char> destination, out int charsWritten)
```

Comment:
nit: you can add remarks tag to tell that charsWritten can be only zero or 12.

Reply:
This is another nonsense in the design of this API. Why does the API return charWritten and bool when it is *always* going to produce 12 characters?

Reply:
> Why does the API return charWritten and bool when it is always going to produce 12 characters?

While it is awkward, is it really that much better to manually put that value in? What if we actually wanted to change this in the future (even via a compat switch)?

@ahsonkhan Do we have any other examples of TryGet where the written chars is always 0 or some other value?

Reply:
> What if we actually wanted to change this in the future (even via a compat switch)?

If we think that the behavior of this API will need tweaking via compat switches in the future, we should rather fix the API design now to make such tweaks unnecessary.